core/modules/system/system.module

```diff
@@ -3112,12 +3116,19 @@ function system_header_block($settings) {
  * @see system_block_view()
  */
 function system_breadcrumb_block($settings = array()) {
-  $crumbs = backdrop_get_breadcrumb();
-  if (isset($settings['current']) && $settings['current'] == TRUE) {
-    $crumbs[] = '<span>' . backdrop_get_title() . '</span>';
-  }
+  // Render breadcrumbs only if the site is not in maintenance mode or the
+  // logged in user has the 'access site in maintenance mode' permission.
+  if (!state_get('maintenance_mode', FALSE) || user_access('access site in maintenance mode')) {
```

Comment:
This check will cause the breadcrumb to disappear within the admin pages as well, where it can be quite helpful even if the site is offline.